manager/allocator/allocator_test.go

```diff
@@ -84,7 +82,7 @@ func TestAllocator(t *testing.T) {
 
 	// Now verify if we get network and tasks updated properly
 	watchNetwork(t, netWatch, false, isValidNetwork)
-	watchTask(t, taskWatch, false, isValidTask)
+	watchTask(t, s, taskWatch, false, isValidTask)
```

Comment:
The broadcaster inside the `MemoryStore` needs to be closed to avoid a memory leak. This means we shouldn't define the `MemoryStore` as a global variable, so it has a well-defined lifetime.